### ISSUE TITLE
Influx parser fails to parse but returns no error

### ISSUE THREAD

**Issue by phemmer** [influxdata/telegraf#2297]
## Bug report

```go
package main

import (
	"fmt"

	"github.com/influxdata/telegraf/plugins/parsers"
)

func main() {
	p, _ := parsers.NewInfluxParser()
	m, err := p.Parse([]byte("test,foo=bar v=1i 123456789"))
	fmt.Printf("err=%v\n", err)
	fmt.Printf("m=%#v\n", m)
}
```

### Expected behavior:

```
err=<nil>
m=[]telegraf.Metric{(*metric.metric)(0xc42009a480)}
```

### Actual behavior:
```
err=<nil>
m=[]telegraf.Metric{}
```

### Additional info:

It parses fine only if there is a trailing newline.

This is a regression. I noticed this while trying to update PR #2094. Code that was previously working is now failing. While I would argue that a trailing newline should not be required to parse, at the very least if it does not parse, it should return an error.

**Comment by sparrc** [comment]
Can you confirm that InfluxDB accepts metrics without newlines (besides via the HTTP service, which in telegraf is also still adding a newline)

**Comment by phemmer** [comment]
Not sure what you mean by "besides via the HTTP service". Why wouldn't that be a valid case of InfluxDB not needing a newline?

But in the case of UDP, yes, it accepts points without newlines:

```
# echo -n 'test,foo=bar v=1i 123456789' | socat - udp-send:localhost:8089

# influx -database telegraf -execute 'select * from test'
name: test
----------
time		foo	v
123456789	bar	1
```

**Comment by sparrc** [comment]
well, the problem really is that we haven't ever defined out line-protocol data format very well. A newline is supposed to be required but apparently we don't enforce that very consistently.

I'll try to get a fix in for that and cherry-pick it into 1.2
